accepted/0000-payg-cloud-integration.md

```diff
@@ -0,0 +1,242 @@
+
+we can call such a zypper plugin ("impersonating zypper") to compute URLs and headers in a trivial and cross-Cloud way.
+https://gist.github.com/moio/b064c1d8cb91a00fd4545f3625ee3911
+
+#### HTTP auth credentials
+Plain HTTP authentication credentials are required to access the repositories.
+Those credentials are stored in plain text `/etc/zypp/credentials.d/<repo_name>` and can be read by root.
+
+#### RMT Server IP
+RMT server domain name are not registered in DNS server, and the IP is specified on the `/etc/hosts` file of the pay-as-you-go instance.
+We can call the command `getent hosts <RMT_HOST_NAME>` to retrieve the IP address, which will cope with future changes to this mechanism.
+
+#### RMT Certificate
+All pay-as-you-go instances connect to RMT servers via https, and their certificates are signed by SUSE, and added to the pay-as-you-go instance at creation.
+We can load the certificate from `/usr/share/pki/trust/anchors/registration_server_<RMT_IP_REPLACE_DOT_WITH_UNDERSCORE>.pem`.
+
+### Remotely run data extraction script
+
+Uyuni sever should open an ssh connection to the pay-as-you-go machine, execute the data extraction script, and retrieve the result in JSON format.
+Uyuni should have support to receive the needed ssh parameters in the web UI (similar to the system bootstrap page) or via the XML-RPC API.
+SSH authentication with basic auth and client certificate should be possible.
+
+For this implementation we will use JSCH library, similar to what is being used in `SSHPushWorker` class.
+
+### Store authentication data from pay-as-you-go instance
+
+Uyuni will create vendor channels for the pay-as-you-go integration. The advantage would be that if such vendor channels had the correct channel label, then mgr-sync would link them to the appropriate products at next `mgr-sync` time, allowing CVE Audit, Product Migration and other features requiring correct product data to work correctly.
+
+Uyuni always gets all products meta information from SCC: which products exists and all repositories assigned.
+These products are only showed in the products setup wizard page if an authentication mechanism for the repositories is available, meaning users have access to it.
+
+The proposed solution will implement a new authentication mechanism on uyuni (named `cloudrmt`, for example) to deal with cloud RMT server authentication. With this approach we will be able to reuse the existing product/channel management features.
+
+#### Implementation step - Cloud RMT server access
+
+To access the cloud RMT server uyuni server needs to know is IP address (which is not registered in DNS) and trust the server certificate.
+
+The current Public Cloud setup requires changing `/etc/hosts` to reach the correct RMT server. This might not be possible if ever Uyuni is delivered as containers. In the constext of this RFC implementation we will follow the same approach a update `/etc/hosts` on Uyuni server.
```

Comment:
Please cite this concern in "drawbacks" below

Reply:
Done